src/Microsoft.Azure.WebJobs.Host/Filters/FunctionInvocationContext.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Filters
+{
+    /// <summary>
+    /// Base context class for <see cref="IFunctionInvocationFilter"/> context objects.
+    /// </summary>
+    public abstract class FunctionInvocationContext : FunctionFilterContext
+    {
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="arguments">The function arguments.</param>
+        /// <param name="properties">The property bag that can be used to pass information between filters.</param>
+        /// <param name="functionInstanceId">The instance ID for the function invocation.</param>
+        /// <param name="functionName">The name of the function.</param>
+        /// <param name="logger"><see cref="ILogger"/> that can be used by the filter to log information.</param>
+        protected FunctionInvocationContext(IReadOnlyDictionary<string, object> arguments, IDictionary<string, object> properties, Guid functionInstanceId, string functionName, ILogger logger)
+            : base(functionInstanceId, functionName, logger)
+        {
+            Arguments = arguments;
+            Properties = properties;
+        }
+
+        /// <summary>
+        /// Gets the function arguments.
+        /// </summary>
+        public IReadOnlyDictionary<string, object> Arguments { get; }
+
+        /// <summary>
+        /// Gets the property bag that can be used to pass information between filters.
+        /// </summary>
+        public IDictionary<string, object> Properties { get; }
```

Comment:
I'm thinking I need to move this to the base FunctionFilterContext so exception filters also have access to this property bag. I think all filters should have access to the shared bag for the invocation scope. In ASP.NET, this can be done via ActionContext.HttpContext.Items which is available to all filter types